apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -1,21 +1,36 @@
 import { Injectable } from '@nestjs/common';
+
+import { SelectIntegration, SelectIntegrationCommand } from '@novu/application-generic';
+import { ChannelTypeEnum } from '@novu/shared';
 import { IntegrationEntity } from '@novu/dal';
-import { GetDecryptedIntegrations, GetDecryptedIntegrationsCommand } from '@novu/application-generic';
 
-import { GetIntegrationsCommand } from '../get-integrations/get-integrations.command';
+import { GetActiveIntegrationsCommand } from './get-active-integration.command';
 
 @Injectable()
 export class GetActiveIntegrations {
-  constructor(private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations) {}
+  constructor(private selectIntegration: SelectIntegration) {}
+
+  async execute(command: GetActiveIntegrationsCommand): Promise<IntegrationEntity[]> {
+    const channelTypes = Object.values(ChannelTypeEnum);
 
-  async execute(command: GetIntegrationsCommand): Promise<IntegrationEntity[]> {
-    return await this.getDecryptedIntegrationsUsecase.execute(
-      GetDecryptedIntegrationsCommand.create({
-        environmentId: command.environmentId,
-        organizationId: command.organizationId,
-        active: true,
-        userId: command.userId,
+    const integrations = await Promise.all(
+      channelTypes.map(async (channelType) => {
+        return await this.selectIntegration.execute(
+          SelectIntegrationCommand.create({
+            environmentId: command.environmentId,
+            organizationId: command.organizationId,
+            userId: command.userId,
+            channelType: channelType,
+            providerId: command.providerId,
+          })
+        );
```

Comment:
@djabarovgeorge we want to return all active integrations, and then having those active we want to mark the ones that will be picked when sending the message, so it should be like this:
```
[
{ name: 'SendGrid Old', channel: 'email', active: true, picked: false }, 
{ name: 'SendGrid New', channel: 'email', active: true, picked: true },
{ name: 'Mailgun', channel: 'email', active: true, picked: false },
{ name: 'Nexmo', channel: 'sms', active: true, picked: false },
{ name: 'Twilio', channel: 'sms', active: true, picked: true }
]
```
So like only one will be picked per channel from active integrations. Please find a better name than `picked`.

Reply:
got you, i thought only the active integration will do the work, my mistake.

do we need the `active:true` in the response object isn't it obvious because it is a get active integration usecaase

Reply:
I think that we should return the same interface as before + additional flag, if we will return the minimized interface then it will be more breaking change

Reply:
awesome that what i made as the initial state will merge shortly

Reply:
One question this PR is part of the transition to adapt the current system or is the final multi provider version? Considering the API mentioned here with the active/picked is the first one but to double check.

Reply:
hard to say if it is the final stage, yes we will reuse the SelectIntegration Use-case and it will work with the current logic, but IMO we will need to revisit this one in the next iterations.